public/controllers/management/components/management/configuration/log-collection/log-collection-windowsevents.js

```diff
@@ -0,0 +1,130 @@
+import { settingsListBuilder } from '../utils/builders';
+import helpLinks from './help-links';
+import { LOGCOLLECTOR_LOCALFILE_PROP, LOCALFILE_WINDOWSEVENT_PROP } from './types'
+
+/**
+ *  Return input label based on logformat value
+ * 
+ * @param {*} value => field value
+ * @param {*} item => settings item
+ * @param {*} config => all log data
+ * @returns string => value to show in Channel label
+ */
+const channelLabel = (value, item, config) => {
+  return  config.logformat === 'eventlog' ? 'Log' :
+          config.logformat === 'eventchannel' ? 'Channel' : 'Channel';
+}
+
+/**
+ * 
+ * @param {*} data => all log data 
+ * @returns string => value to show in query input
+ */
+const queryValue = (data) => { 
+  return  typeof data === 'undefined' ? '-' : 
+          typeof data === 'object' ? data.value : data;
+}
+
+/**
+ * Returns targets array parsed in one string
+ * @param {*} item 
+ * @returns string => target
+ */
+const renderTargetField = item => item ? item.join(', ') : 'agent';
+/**
+ * Return panels title
+ * @param {*} item => log data
+ * @returns 
+ */
+const panelsLabel = (item) => !item.channel ? `${item.logformat} - ${renderTargetField(item.target)}` : `${item.channel} (${item.logformat})`;
+
+const mainSettings = [
+  { field: 'logformat', label: 'Log format' },
+  { field: 'channel', label: 'Channel', render: renderValueOrNoValue, renderLabel: channelLabel },
+  { field: 'query', label: 'Query', render: queryValue },
+  {
+    field: 'only-future-events',
+    label: 'Only future events',
+    render: renderValueOrNoValue
+  },
+  {
+    field: 'reconnect_time',
+    label: 'Reconnect Time',
+    render: renderValueOrNoValue
+  }
+];
+
+class WzConfigurationLogCollectionWindowsEvents extends Component {
+  constructor(props) {
+    super(props);
+  }
+  render() {
+    const { currentConfig } = this.props;
+    const items =
+      currentConfig[LOGCOLLECTOR_LOCALFILE_PROP] &&
+      currentConfig[LOGCOLLECTOR_LOCALFILE_PROP][LOCALFILE_WINDOWSEVENT_PROP]
+        ? settingsListBuilder(currentConfig[LOGCOLLECTOR_LOCALFILE_PROP][LOCALFILE_WINDOWSEVENT_PROP], panelsLabel)
+        : [];
+    return (
+      <Fragment>
+        {currentConfig[LOGCOLLECTOR_LOCALFILE_PROP] &&
+          isString(currentConfig[LOGCOLLECTOR_LOCALFILE_PROP]) && (
+            <WzNoConfig
+              error={currentConfig[LOGCOLLECTOR_LOCALFILE_PROP]}
+              help={helpLinks}
+            />
+          )}
+        {currentConfig[LOGCOLLECTOR_LOCALFILE_PROP] &&
+        !isString(currentConfig[LOGCOLLECTOR_LOCALFILE_PROP]) &&
+        !(currentConfig[LOGCOLLECTOR_LOCALFILE_PROP][LOCALFILE_WINDOWSEVENT_PROP] || [])
+          .length ? (
+          <WzNoConfig error="not-present" help={helpLinks} />
+        ) : null}
+        {currentConfig[LOGCOLLECTOR_LOCALFILE_PROP] &&
+        !isString(currentConfig[LOGCOLLECTOR_LOCALFILE_PROP]) &&
+        currentConfig[LOGCOLLECTOR_LOCALFILE_PROP][LOCALFILE_WINDOWSEVENT_PROP] &&
```

Comment:
This kind of conditional can be replaced by the optional chaining operator. ?. 
It would simplify the conditional and improve readability.
e.g.
`currentConfig?.[LOGCOLLECTOR_LOCALFILE_PROP]?.[LOCALFILE_WINDOWSEVENT_PROP]?.length`